api/v1beta2/backend_types.go

```diff
@@ -0,0 +1,413 @@
+/*
+Copyright 2021 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	crossplanetypes "github.com/oam-dev/terraform-controller/api/types/crossplane-runtime"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:storageversion
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="STATE",type="string",JSONPath=".status.state"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+
+// Backend is the Schema for the backend API.
+type Backend struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec BackendSpec `json:"spec,omitempty"`
+}
+
+// BackendSpec defines the desired state of Backend
+type BackendSpec struct {
+	// Backend contains all options supported by the Terraform backend configuration.
+	Backend BackendSpecInnerMain `json:"backend,omitempty"`
+}
+
+// BackendSpecInnerMain defines all options supported by the Terraform backend configuration.
+type BackendSpecInnerMain struct {
```

Comment:
What's the differences of creating a new Backend CRD between expand spec.Backend of Configuration?

Reply:
Creating a new `Backend` CRD will make the `Configuration` more simple and we can also add more fields like options describing `backup and resotre` to the new CRD.

Reply:
As one Configuration will match one Backend, so you can also describe backup in Configuration. The CRD yamls will get longer, but for end-users, it won't as they will only choose one backend.
Besides, maintaining another Object will need to take care of its deletion.

Reply:
Okay. I will move the spec into the `Configuratoin`.